crates/index-scheduler/src/test_utils.rs

```diff
@@ -349,4 +390,18 @@ impl IndexSchedulerHandle {
         }
         self.advance_till([AfterProcessing]);
     }
+
+    // Wait for one failed batch.
+    #[track_caller]
+    pub(crate) fn scheduler_is_down(&mut self) {
+        loop {
+            match self
+            .test_breakpoint_rcv
+            .recv_timeout(std::time::Duration::from_secs(1)) {
+                Ok((_, true)) => continue,
+                Ok((b, false)) => panic!("The scheduler was supposed to be down but successfully moved to the next breakpoint: {b:?}"),
+                Err(RecvTimeoutError::Timeout | RecvTimeoutError::Disconnected) => break,
```

Comment:
does that work with just disconnected instead of the timeout?

Reply:
I checked, and no, since the index-scheduler receiver channel has not been dropped it can still answer with `Timeout` :pensive: